dot/network/block_announce.go

```diff
@@ -183,6 +188,10 @@ func (s *Service) validateBlockAnnounceHandshake(from peer.ID, hs Handshake) err
 		return errors.New("invalid handshake type")
 	}
 
+	if int(bhs.Roles) > 4 {
```

Comment:
`3` is an invalid role, right? should you include an error for it?

Reply:
We should use a typed byte and constants to be sure about role values etc. (and have meaningful constant names too to know which is which). Dealing with numbers like this is just sus

Reply:
addressed